Added 404 Page, with proper response

RFX14 opened the pull request:
Resolves #14 open to different design suggestions

RFX14 committed:
Added correct page linking

RFX14 committed:
Added 404 Page, with proper response

RFX14 committed:
Added 404 Page, with proper response

calvh committed:
Add test for "/" route

Add pytest to requirements.txt

calvh commented on app/__init__.py:
```diff
@@ -12,7 +12,11 @@
-@app.route('/', defaults={'person':'Miyabi'})
+
+@app.errorhandler(404)
+def not_found(e):
```
I prefer to use the full word error instead of  `e` since sometimes when we deal with frontend we have events as well and that could be an `e` too. What do you think?

calvh commented on app/static/styles/main.css:
```diff
@@ -129,4 +129,42 @@ li a:hover:not(.active) {
 
 .active {
   background-color: #30519C;
+}
```
Since these are specific to the 404 page, maybe we could make it a separate file?

calvh commented on app/templates/404.html:
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
```
Would it be possible to use the `{% extends 'base.html' %}` syntax to avoid rewriting the first 20 lines?

calvh replied:
Actually, I'm not sure that's a good idea. Maybe we should make it so that the navbar doesn't appear in the 404. What do you think?

RFX14 replied:
I don't think having a navbar would look good on the 404 page, however maybe we could just put a simple link back to the home page. Thoughts?

calvh replied:
Great idea let's do that

calvh reviewed (changes requested):
Great work! Just some minor comments

calvh committed:
Add Testing section

calvh committed:
Fix render functions in __init__.py

Rename index.html to base.html
Modified layout of base.html
Add index.html as template for the About page
Add templates for contact, resume, projects pages
Fix CSS selector for ul tags

calvh committed:
Merge pull request #15 from calvh/9-page-linking

Added correct page linking

RFX14 committed:
Merge pull request #17 from calvh/13-test-root-url

Add test for "/" route

RFX14 committed:
Merge pull request #22 from calvh/21-update-readme

Add Testing section

RFX14 committed:
Added some fixes that were suggested

RFX14 committed:
Implemented fixes suggested

RFX14 committed:
Added scaling to 404

RFX14 committed:
Added home button in 404 page

calvh committed:
Change background color of 404 page

calvh commented:
Fantastic work @RFX14
